data-prepper-plugins/parse-json-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/parsejson/ParseJsonProcessorTest.java

```diff
@@ -287,6 +288,24 @@ void test_when_condition_skips_processing_when_evaluates_to_false() {
 
     }
 
+    @Test
+    void test_tags_when_json_parse_fails() {
```

Comment:
What happens if tags don't exist? Can we add a check to ensure we aren't building tag set will null values?

Reply:
@cmanning09 you mean, I should check to make sure tags are null when no failures? Something like `assertThat(parsedEvent.getMetadata().getTags(), equalTo(Set.of()));` to be added in all other tests?

Reply:
Yes that works. Feel free to add this in a follow up PR.